src/lib/eliom_registration.server.ml

```diff
@@ -107,29 +107,26 @@ module Make_typed_xml_registration
   (Typed_xml: Xml_sigs.Typed_xml with module Xml := Xml)
   (E : sig type content end) = struct
 
-    module Format = Xml_print.Make_typed(Xml)(Typed_xml)(Ocsigen_stream.StringStream)
+    module Fmt = Xml_print.Make_typed_fmt(Xml)(Typed_xml)
 
-    let result_of_content_subxhtml get_etag c =
+    let out =
       let encode x = fst (Xml_print.Utf8.normalize_html x) in
-      let x = Format.print_list ~encode c in
+      Format.asprintf "%a" (Fmt.pp_elt ~encode ())
+
+    let out_list l =
+      List.map out l
+      |> String.concat ""
+      |> Ocsigen_stream.StringStream.put
+      |> Ocsigen_stream.StringStream.make
```

Comment:
`Format.pp_print_list ~sep:(fun _ () -> ())` to output lists.

Morally, we should get rid of StringStream and provide a buffer-like API (which would make creation of a formatter easy), but in the meantime, you can probably make a formatter out of the put function (or just don't bother for now).